pkg/sql/schemachanger/scbuild/builder_state.go

```diff
@@ -600,6 +609,8 @@ func (b *builderState) resolveRelation(
 	}
 	err, found := c.privileges[p.RequiredPrivilege]
 	if !found {
+		// Validate if this descriptor can be resolved under the current schema.
```

Comment:
this smells to me. Shouldn't the `MayResolveTable` be taking this into consideration? Like if we don't have usage, then we need to look in the next schema. Consider:

As admin:
```sql
CREATE SCHEMA sc1;
CREATE SCHEMA sc2;
CREATE USER new_user;
GRANT CREATE, USAGE ON SCHEMA sc1 TO new_user;
GRANT CREATE, USAGE ON SCHEMA sc2 TO new_user;
```

```sql
CREATE TABLE sc1.foo (i INT PRIMARY KEY);
CREATE TABLE sc2.foo (i INT PRIMARY KEY);
```

admin
```sql
REVOKE CREATE, USAGE ON sc1 FROM new_user;
```

new_user:
```
SET search_path = sc1, sc2;
ALTER TABLE foo ADD COLUMN j INT;
```
The above should alter `sc2.foo`.

Reply:
cc @chengxiong-ruan as you've been playing here too